### PR TITLE
SW-5281 Dialog flow for "Add species to project"

### PR DESCRIPTION
**Pull request by constanzauanini**
Screenshot 2024-05-02 at 5 23 42 PM

### CODE REVIEW

File: src/components/SpeciesDeliverableTable/AddSpeciesModal.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    setSelectableSpecies(speciesToAdd);
+  }, [allSpecies]);
+
+  const [record] = useState<ParticipantProjectSpeciesRequest | undefined>(initializeSpeciesProject());
+
+  const save = () => {
+    if (selectedSpecies) {
+      setError('');
+      // const newProjectSpecies = { ...record, speciesId: selectedSpecies.id };
+      // save newProjectSpecies
```
Comment (constanzauanini) [inline src/components/SpeciesDeliverableTable/AddSpeciesModal.tsx:74]: @joshbuchea @nickgraz I see we have the selectParticipantProjectSpeciesUpdateRequest created, but don't we need a selectParticipantProjectSpeciesCreateRequest?

Comment (nickgraz) [reply src/components/SpeciesDeliverableTable/AddSpeciesModal.tsx:74]: Hmm that is correct. Not sure how I missed that

Comment (nickgraz) [reply src/components/SpeciesDeliverableTable/AddSpeciesModal.tsx:74]: Are you able to add it?

Comment (constanzauanini) [reply src/components/SpeciesDeliverableTable/AddSpeciesModal.tsx:74]: Sure, I will add it.

**Review by nickgraz — APPROVED** [review]
Looks good to me!

**Comment by joshbuchea** [comment]
Sorry about the conflicts 🙇 This looks great!

**Pull request merged**